src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs

```diff
@@ -247,22 +247,28 @@ internal UnitTestResult[] RunTestMethod()
                                     watch.Start();
 
                                     this.testContext.SetDataRow(dataRow);
-                                    UTF.TestResult currentResult;
+                                    UTF.TestResult[] testResults;
 
                                     try
                                     {
-                                        currentResult = this.testMethodInfo.TestMethodOptions.Executor.Execute(this.testMethodInfo)[0];
+                                        testResults = this.testMethodInfo.TestMethodOptions.Executor.Execute(this.testMethodInfo);
                                     }
                                     catch (Exception ex)
                                     {
-                                        currentResult = new UTF.TestResult() { TestFailureException = new Exception(string.Format(CultureInfo.CurrentCulture, Resource.UTA_ExecuteThrewException, ex.Message), ex) };
+                                        testResults = new[]
+                                        {
+                                            new UTF.TestResult() { TestFailureException = new Exception(string.Format(CultureInfo.CurrentCulture, Resource.UTA_ExecuteThrewException, ex.Message), ex) }
+                                        };
                                     }
 
-                                    currentResult.DatarowIndex = rowIndex++;
                                     watch.Stop();
-                                    currentResult.Duration = watch.Elapsed;
+                                    foreach (var testResult in testResults)
+                                    {
+                                        testResult.DatarowIndex = rowIndex++;
```

Comment:
DatarowIndex is used for each row of data in dataSource. 
By framework extensibility you have now started running each dataRow multiple times(say 5), but we should not increase dataRowIndex for each of the 5 runs of a DataRow. It should be incremented only once per dataRow and not once per testResult.
This statement should be testResult.DataRowIndex = rowIndex;
Do rowIndex++ separately outside of the loop.